### PR TITLE
add automatically generated sass variable documentation page to NDS

### LINKED ISSUE #2214: Implement SASS Docs for component pages in Gatsby
**User Story:**
As Spark we want to implement SASS docs into the Gatsby Site so that we will have a single page with all the sass variables displayed 

**Notes:**
This is the Implementation from Research Issue: #1439
SASS DOC: a documentation system to build pretty and powerful docs in the blink of an eye
  
**AC:**
- Review #1439 
- Implement SASS doc
- Will have 1 page with all the sass variables will be there
- pages will live in gatsby
- write comments in all our sass files

### PR DESCRIPTION
**Pull request by afebbraro**
## What does this PR do?
Adds automatically generated sass variable documentation to the new docs site. It uses the sass doc package (http://sassdoc.com/annotations/). 

### Associated Issue 
Fixes #2214 

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR
then please remove it.

### Documentation
 - [x] Add new sass var page to gatsby

### Code
 - [x] Build Component in Spark Vanilla (Sass, HTML, JS)
 - [x] Unit Testing in Spark Vanilla with `npm run test` (100% coverage, 100% passing)
 - [x] Unit Testing in Spark Angular with `gulp test-angular` (100% coverage, 100% passing)
 - [x] Unit Testing in Spark React with `gulp test-react` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### Design Review
 - [ ] Design reviewed and approved --- we still need to design the Sass Variables page in gatsby. That will be done in another issue.

### CODE REVIEW

File: html/settings/_settings.scss [hunk 1]
```diff
@@ -5,920 +5,1768 @@
...
+/// The font weight of TypeBodyThree.
 $sprk-font-weight-body-three: 300 !default;
 
 // Body Four
+/// The font-family, including fallbacks of TypeBodyFour.
 $sprk-font-family-body-four: RocketSans, Helvetica, 'Helvetica Neue', Arial,
   sans-serif !default;
+/// The font size of TypeBodyFour.
 $sprk-font-size-body-four: 0.8125rem !default;
+/// The color of the TypeBodyFour text.
```
Comment (RVMendoza) [inline html/settings/_settings.scss]: nit picky things: do we need to add "text" to this? the other lines don't have it.

Comment (afebbraro) [reply html/settings/_settings.scss]: I removed it, thanks!

File: html/settings/_settings.scss [hunk 2]
```diff
@@ -5,920 +5,1768 @@
...
+/// The maximum width of the Modal.
 $sprk-modal-max-width: 43.75rem !default;
+/// The maximum height of the Modal.
 $sprk-modal-height: 70% !default;
+/// The point at which the Modal
+/// moves farther down the viewport from the top.
 $sprk-modal-breakpoint-small: 25rem !default;
+/// The point at which the Modal moves even
+/// farther down the viewport from the top than
+// it does at the small breakpoint.
```
Comment (RVMendoza) [inline html/settings/_settings.scss]: Simplified:
Determines how far down from the top of the page the Modal renders at XXX breakpoint.

File: html/settings/_settings.scss [hunk 3]
```diff
@@ -5,920 +5,1768 @@
...
+/// items in the content section in the Masthead on wide viewports.
 $sprk-masthead-content-item-padding-right-wide: $sprk-space-m !default;
+/// The padding left of the content
+/// items in the content section in the Masthead on wide viewports.
 $sprk-masthead-content-item-padding-left-wide: $sprk-space-m !default;
+/// The visited state color of Masthead links.
 $sprk-masthead-link-visited-color: $sprk-black !default;
+/// The hover state color of Masthead links.
 $sprk-masthead-link-hover-color: $sprk-green !default;
+/// The font weight of Masthead links.
```
Comment (RVMendoza) [inline html/settings/_settings.scss]: for -> of

Comment (RVMendoza) [inline html/settings/_settings.scss]: Do we want to capitalize the `S` in Stepper Step? Since it is an actual component.

Comment (afebbraro) [reply html/settings/_settings.scss]: I only capitalized the components in the DS. Technically on the backend, we have made the step a component too but its not something we have documented as an available component in the DS on its own.

**Pull request merged**